clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -2325,6 +2325,7 @@ void CodeGenModule::SetLLVMFunctionAttributesForDefinition(const Decl *D,
       B.addAttribute(llvm::Attribute::Naked);
 
     // OptimizeNone wins over OptimizeForSize and MinSize.
+    F->removeFnAttr(llvm::Attribute::OptimizeForDebugging);
```

Comment:
Given that this patch doesn't have any other clang changes, this one can be deferred to a later patch where clang actually applies the attribute.

Reply:
+1, and otherwise the change makes sense to me. Could you post about this on Discourse, pointing to this pull request - just to make sure folks are aware we're adding a new IR attribute here? Be good to ensure there's consensus.

Reply:
Topic posted: https://discourse.llvm.org/t/rfc-new-function-attribute-optdebug-for-prioritizing-debuggability-in-llvm-passes/73552